pkg/controllers/clusteroperator_controller.go

```diff
@@ -110,7 +105,38 @@ func (r *CloudOperatorReconciler) Reconcile(ctx context.Context, _ ctrl.Request)
 		return ctrl.Result{}, err
 	}
 
-	if err := r.sync(ctx, operatorConfig); err != nil {
+	// Get resources for platform the specific platform
+	resources, err := cloud.GetResources(operatorConfig)
+	if err != nil {
+		klog.Errorf("Unable to render operands manifests: %s", err)
+		if err := r.setStatusDegraded(ctx, err); err != nil {
+			klog.Errorf("Error syncing ClusterOperatorStatus: %v", err)
+			return ctrl.Result{}, fmt.Errorf("error syncing ClusterOperatorStatus: %v", err)
+		}
+	}
+
+	allowedToProvision, err := r.provisioningAllowed(ctx, infra)
+	if err != nil {
+		klog.Errorf("Unable to determine cluster state to check if provision is allowed: %v", err)
+		return ctrl.Result{}, err
+	} else if !allowedToProvision {
+		operandsProvisioned, err := r.operandsProvisioned(ctx, resources)
+		if err != nil {
+			klog.Errorf("Unable to determine if operands are already provisioned: %v", err)
+		}
+		if operandsProvisioned {
+			if err := r.deleteOperands(ctx, resources); err != nil {
+				if err := r.setStatusDegraded(ctx, err); err != nil {
```

Comment:
will we need to become un-degraded once all the resources are removed?

Reply:
On next reconciliation this status will be set to available during the check in the `provisioningAllowed ` procedure

Reply:
https://github.com/openshift/cluster-cloud-controller-manager-operator/pull/238/files#diff-1e56759e83cb848f28cc71546c3151724a2bc26915e59e279ab8f702858a91d9R327